File: generators/app/templates/middle-tier/src/main/kotlin/javalin/service/UserServiceImpl.kt
package <%= packageName %>.service

import <%= packageName %>.domain.User
import <%= packageName %>.repository.MemoryUserRepository

class UserServiceImpl : UserService {

    // Use static singleton component instances for services and repositories, or any dependency injection framework
    private val userRepository = MemoryUserRepository()

    override fun users(): Collection<User> {
        return userRepository.readUsers()
    }

    override fun user(username: String): User? {
        return userRepository.readUser(username)
    }

}

File: generators/app/templates/middle-tier/src/main/scala/spring-boot/service/UserServiceImpl.scala
package <%= packageName %>.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

import <%= packageName %>.domain.User
import <%= packageName %>.repository.UserRepository

@Service
class UserServiceImpl extends UserService {

    @Autowired
    private val userRepository: UserRepository = null

    override def users: List[User] = userRepository.readUsers

    override def user(username: String): Option[User] = userRepository.readUser(username)

}
